src/shared/nsresource.c

```diff
@@ -32,10 +33,18 @@ static int make_pid_name(char **ret) {
         char spid[DECIMAL_STR_MAX(pid_t)];
         xsprintf(spid, PID_FMT, getpid_cached());
 
-        assert(strlen(spid) <= 16);
-        strshorten(comm, 16 - strlen(spid));
+        /* Include a counter in the name, so that we can allocate multiple namespaces per process, with
+         * unique names. For the first namespace we suppress the suffix */
+        char scounter[sizeof(counter) * 2 + 1];
+        if (counter == 0)
+                scounter[0] = 0;
+        else
+                xsprintf(scounter, "%" PRIx64, counter);
+        counter++;
 
-        _cleanup_free_ char *s = strjoin(comm, spid);
+        strshorten(comm, LESS_BY(LESS_BY(16U, strlen(spid)), strlen(scounter)));
```

Comment:
Maybe better to use `TASK_COMM_LEN` rather than 16.

Reply:
the 16 is the namespace max length, not the comm max length. the comment above should have made that clear...

but I'll add a define for the limit, to make this clearer.